File: app/src/main/java/dreammaker/android/expensetracker/viewmodel/PeopleListViewModel.java
package dreammaker.android.expensetracker.viewmodel;

import android.app.Application;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import dreammaker.android.expensetracker.database.dao.PersonDao;
import dreammaker.android.expensetracker.database.model.PersonModel;

@SuppressWarnings("unused")
public class PeopleListViewModel extends DBViewModel {

    private LiveData<List<PersonModel>> mPeople;

    public PeopleListViewModel(@NonNull Application application) {
        super(application);
    }

    @NonNull
    public PersonDao getPersonDao() {
        return getExpenseDatabase().getPersonDao();
    }

    @NonNull
    public LiveData<List<PersonModel>> getAllPeople() {
        if (null == mPeople) {
            mPeople = getPersonDao().getAllPeopleLive();
        }
        return mPeople;
    }
}

File: app/src/main/java/dreammaker/android/expensetracker/ui/person/personlist/PeopleListViewModel.kt
package dreammaker.android.expensetracker.ui.person.personlist

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import dreammaker.android.expensetracker.database.ExpensesDatabase
import dreammaker.android.expensetracker.database.PersonDao
import dreammaker.android.expensetracker.database.PersonModel

class PeopleListViewModel(app: Application): AndroidViewModel(app) {

    private val personDao: PersonDao

    init {
        val db = ExpensesDatabase.getInstance(app)
        personDao = db.personDao
    }

    private lateinit var people: LiveData<List<PersonModel>>

    fun getAllPeople(): LiveData<List<PersonModel>> {
        if (!::people.isInitialized) {
            people = personDao.getAllPeople()
        }
        return people
    }
}